### ISSUE TITLE
group with `nan` values is strange

### ISSUE THREAD

**Issue by yifanwu** [data-8/datascience#416]
If you load the following csv file and attempt to do group by on county, it treats each `nan` as unique---I think the nans should be treated as the same value. Happy to be convinced otherwise.

```
county,price,year
,17899,2012.0
,0,2016.0
,46463,2015.0
,0,2016.0
,49999,2018.0
,13999,2009.0
,34500,2017.0
```

**Comment by adnanhemani** [comment]
Hi @yifanwu. I took a look - it's reproducible (thank you for the example), here's why: np.float64 NaN objects aren't equal to each other in Python. Here's a SO post with a deeper explanation: https://stackoverflow.com/questions/6441857/nans-as-key-in-dictionaries

That being said, I agree that this isn't the best behavior - I'm going to be putting in a PR for this. Thanks for reporting the issue!

**Comment by adnanhemani** [comment]
Closed by #417. Hope this helps :)

**Comment by yifanwu** [comment]
Thanks Adnan for the timely fix!
